userbenchmark/cpu/run_config.py

```diff
@@ -43,21 +40,19 @@ def dump_result_to_json(metrics, output_dir):
         result = get_output_json(BM_NAME, metrics)
         dump_output(BM_NAME, result, output_dir)
 
-    def validate(candidate: str, choices: List[str]) -> str:
-        """Validate the candidates provided by the user is valid"""
-        assert candidate in choices, f"Specified {candidate}, but not in available list: {choices}."
-        return candidate
-
-    def run_config(config: TorchBenchModelConfig, dryrun: bool=False) -> Optional[TorchBenchModelMetrics]:
+    def run_config(config: TorchBenchModelConfig, metrics: List[str], dryrun: bool=False) -> Optional[TorchBenchModelMetrics]:
         """This function only handles NotImplementedError, all other errors will fail."""
-        metrics = get_metrics(config)
         print(f"Running {config} ...", end='')
         if dryrun:
             return None
         # We do not allow RuntimeError in this test
         try:
-            # load the model instance within separate subprocess
-            model = load_model_isolated(config)
+            if "cpu_peak_mem" in metrics:
+                # load the model instance within separate subprocess
+                model = load_model_isolated(config)
+            else:
+                # load the model instance within current process
+                model = load_model(config)
```

Comment:
Curious why using `load_model` here?
We should try to avoid loading model instance within the same process. Different model code might interfere with each other, so it will be better to isolate them by running models in separate processes.

Reply:
The mainly reason is that the `load_model_isolated` can't show verbose / log when we set the `ONEDNN_VERBOSE` or `PYTORCH_JIT_LOG_LEVEL` and other verbose / log flags. Those flags can help us to double check whether we run into right pass or get more debug information. How do you think?